subt/docker/robotika/run_solution.bash

```diff
@@ -13,35 +13,28 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # enable ROS DEBUG output to see if messages are being dropped
 export ROSCONSOLE_CONFIG_FILE="${DIR}/rosconsole.config"
 
+# when signal received just exit
+trap "exit;" HUP INT TERM
+
+# when exiting, send SIGINT to all processes in current process group
+# and wait for them to finish
+trap "kill -s SIGINT 0; wait" EXIT
+
 echo "Waiting for robot name"
 while [ -z "$ROBOT_NAME" ]; do
     ROBOT_NAME=$(rosparam get /robot_names)
     sleep 0.5
 done
 echo "Robot name is '$ROBOT_NAME'"
 
-echo "Start robot solution"
-export OSGAR_LOGS=/osgar-ws/logs
-/osgar-ws/env/bin/python3 -m subt run /osgar-ws/src/osgar/subt/zmq-subt-x2.json --side auto --walldist 0.8 --timeout 100 --speed 1.0 --note "run_solution.bash" &
-ROBOT_PID=$!
-
-# Run your solution and wait for ROS master
+echo "Starting ros<->osgar proxy"
+# Wait for ROS master, then start proxy
 # http://wiki.ros.org/roslaunch/Commandline%20Tools#line-45
 roslaunch proxy sim.launch --wait robot_name:=$ROBOT_NAME &
-ROS_PID=$!
-
-# Turn everything off in case of CTRL+C and friends.
-function shutdown {
-       kill ${ROBOT_PID}
-       kill ${ROS_PID}
-       wait
-       exit
-}
-trap shutdown SIGHUP SIGINT SIGTERM
 
-
-# Wait for the controllers to finish.
-wait ${ROBOT_PID}
+echo "Starting osgar"
+export OSGAR_LOGS=/osgar-ws/logs
+/osgar-ws/env/bin/python3 -m subt run /osgar-ws/src/osgar/subt/zmq-subt-x2.json --side auto --walldist 0.8 --timeout 100 --speed 1.0 --note "run_solution.bash" &
 
 echo "Sleep and finish"
 sleep 30
```

Comment:
I do not understand how it is guaranteed that script terminates after both ros proxy and osgar end??

Reply:
OK "&" would explain it, thanks - I would like to test it but probably not tonight (i.e. I do not see any other problem in the code now although my understanding is quite limited)